core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -70,7 +70,9 @@ class DelayedOperations(topicPartition: TopicPartition,
   def numDelayedDelete: Int = deleteRecords.numDelayed
 }
 
-object Partition extends KafkaMetricsGroup {
+object Partition {
+  private val metricsGroup = new KafkaMetricsGroup(classOf[Partition])
```

Comment:
I think we can call this and other similar `val`s `metrics`. It's a bit more concise and the `Group` suffix doesn't add any particular value.

Reply:
This works for some classes, but some other (e.g. `KafkaBroker`, `KafkaController`) already have the `metrics` field. Maybe let's keep the suffix for consistency?

Reply:
That's fine.